scripts/cangen/cangen/main.py

```diff
@@ -0,0 +1,45 @@
+"""
+Author: Samuel Parent
+Date: 2024-04-13
+"""
+
+import argparse
+import logging
+import sys
+
+from .can_generator import generate_can_from_dbc
+
+logging.basicConfig(level="INFO", format="%(levelname)-8s| (%(name)s) %(message)s")
+logger = logging.getLogger(__name__)
+
+
+def parse():
+    parser = argparse.ArgumentParser(description="DBC to C code generator")
+    parser.add_argument("project", type=str, help="Name of the project")
+    parser.add_argument(
+        "--log-level",
+        dest="level",
+        choices=["STATUS", "INFO", "VERBOSE", "DEBUG"],
+        default="INFO",
+        help="Log verbosity threshold",
+    )
+
+    # If parsing fails (ex incorrect or no arguments provided) then this exits with
+    # code 2.
+    return parser.parse_args(None if sys.argv[1:] else ["-h"])
```

Comment:
ik I asked you to add the `None if ...` but this violates the comment above which states that the program will exit with code 2 if parsing fails.

Since this tool is primarily used by other scripts, not by humans, I'd prefer the error code instead of the help message. With this setup, there is no way to distinguish success from the failed run where no args are passed.